### PR TITLE
dev container build fixes

### LINKED ISSUE #1167: [BUG] Trying to build latest main devcontainer fails on docker-client
**Describe the bug**
The latest docker group_id code introduced in the devcontainer seems to fail when trying to build the devcontainer. Have tested this across Win10/WSL, Mac and Codespaces all with same result:

```
ERROR: executor failed running [/bin/sh -c /tmp/docker-client.sh $USERNAME]: exit code: 2
[2022-01-24T23:18:37.188Z] ------
2022-01-24T23:18:37:  > [ 7/18] RUN /tmp/docker-client.sh vscode:
2022-01-24T23:18:37: ------
2022-01-24T23:18:37: executor failed running [/bin/sh -c /tmp/docker-client.sh $USERNAME]: exit code: 2
Error: Command failed: docker build -f /var/lib/docker/codespacemount/workspace/AzureTRE/.devcontainer/Dockerfile -t vsc-azuretre-ecbd6b3c66a8c2c9cdd9e501275739ac --build-arg DOCKER_GROUP_ID= /var/lib/docker/codespacemount/workspace/AzureTRE
    at fz (/usr/lib/node_modules/@microsoft/vscode-dev-containers-cli/dist/node/devContainersCLI.js:342:2030)
    at async $ue (/usr/lib/node_modules/@microsoft/vscode-dev-containers-cli/dist/node/devContainersCLI.js:340:8190)
    at async Y2 (/usr/lib/node_modules/@microsoft/vscode-dev-containers-cli/dist/node/devContainersCLI.js:340:4649)
    at async _ue (/usr/lib/node_modules/@microsoft/vscode-dev-containers-cli/dist/node/devContainersCLI.js:336:10937)
    at async _Q (/usr/lib/node_modules/@microsoft/vscode-dev-containers-cli/dist/node/devContainersCLI.js:368:3257)
    at async Hke (/usr/lib/node_modules/@microsoft/vscode-dev-containers-cli/dist/node/devContainersCLI.js:368:22351)
    at async zke (/usr/lib/node_modules/@microsoft/vscode-dev-containers-cli/dist/node/devContainersCLI.js:368:21957)
```

**Steps to reproduce**

1. Open main branch and rebuild devcontainer
1. Build of devcontainer will fail with above message

### LINKED ISSUE #1168: [BUG] dev container build fails on terraform installation
Remember to label the issue with the appropriate area, and also assign it to yourself if you plan to work on the issue in the near future.

**Describe the bug**
Building the dev container results in the following error:

```bash
...
#9 11.57 Reading package lists...
#9 11.99 E: Encountered a section with no Package: header
#9 11.99 E: Problem with MergeList /var/lib/apt/lists/apt.releases.hashicorp.com_dists_buster_main_binary-amd64_Packages.lz4
#9 11.99 E: The package lists or status file could not be parsed or opened.
------
executor failed running [/bin/sh -c apt-get update && apt-get install -y gnupg software-properties-common curl     && curl -fsSL https://apt.releases.hashicorp.com/gpg | apt-key add -     && apt-add-repository "deb [arch=amd64] https://apt.releases.hashicorp.com $(lsb_release -cs) main"     && apt-get update && apt-get install -y terraform "${TERRAFORM_VERSION}"]: exit code: 100
```

**Steps to reproduce**

1. Pull `main` (currently `8db5c9a9cddda7f12784845381be8dfa0c0bc6aa`) and build the dev container (without cache)

### PR DESCRIPTION
**Pull request by stuartleeks**
- [X] switch to releases.hashicorp.com for terraform releases - fixes #1168
- [X] Fixup docker group env var and script - fixes #1167

### CODE REVIEW

**Review by ross-p-smith — APPROVED** [review]

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -7,7 +7,8 @@
 			// To ensure that the group ID for the docker group in the container
 			// matches the group ID on the host, add this to your .bash_profile on the host
 			// 		export DOCKER_GROUP_ID=$(getent group docker | awk -F ":" '{ print $3 }')
-			"DOCKER_GROUP_ID": "${localEnv:DOCKER_GROUP_ID:-1001}"
+			// "DOCKER_GROUP_ID": "${localEnv:DOCKER_GROUP_ID:-1001}"
```
Comment (jjgriff93) [inline .devcontainer/devcontainer.json:10]: May as well remove this

File: .devcontainer/Dockerfile [hunk 2]
```diff
@@ -22,14 +22,12 @@ RUN if [ "${INSTALL_NODE}" = "true" ]; then su $USERNAME -c "umask 0002 && . /us
...
-    && curl -fsSL https://apt.releases.hashicorp.com/gpg | apt-key add - \
-    && apt-add-repository "deb [arch=amd64] https://apt.releases.hashicorp.com $(lsb_release -cs) main" \
-    && apt-get update && apt-get install -y terraform "${TERRAFORM_VERSION}"
+COPY .devcontainer/scripts/terraform.sh /tmp/
+RUN bash /tmp/terraform.sh "${TERRAFORM_VERSION}" /usr/bin
 
-ARG DOCKER_GROUP_ID
-COPY .devcontainer/scripts/docker-client.sh /tmp/
-RUN /tmp/docker-client.sh $USERNAME
+# ARG DOCKER_GROUP_ID
```
Comment (jjgriff93) [inline .devcontainer/Dockerfile:28]: Add these back in? (saves me PRing a small change)

**Review by jjgriff93 — APPROVED** [review]
Given some of my changes are in here may as well add the rest (see my comments) to save a subsequent PR

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (stuartleeks) [reply .devcontainer/devcontainer.json:10]: oops - included by mistake. Lemme add the rest of the docker stuff we discussed to make amends!

File: .devcontainer/Dockerfile (hunk 2 above)
Comment (stuartleeks) [reply .devcontainer/Dockerfile]: oops

**Pull request merged**